src/idiomorph.js

```diff
@@ -1063,11 +1074,48 @@ var Idiomorph = (function () {
         function removeNode(tempNode, ctx) {
             removeIdsFromConsideration(ctx, tempNode)
             if (ctx.callbacks.beforeNodeRemoved(tempNode) === false) return;
-
-            tempNode.parentNode?.removeChild(tempNode);
+            if (ctx.pantry) {
+                moveToPantry(tempNode, ctx);
+            } else {
+                tempNode.parentNode?.removeChild(tempNode);
+            }
             ctx.callbacks.afterNodeRemoved(tempNode);
         }
 
+        function moveToPantry(node, ctx) {
+            // If the node is a leaf (no children), process it, and then we're done
+            if (!node.children || node.children.length === 0) {
+                if (node.id) {
+                    ctx.pantry.appendChild(node);
+                }
+
+            // otherwise we need to process the children first
+            } else {
+                Array.from(node.children).forEach(child => {
+                    moveToPantry(child, ctx);
+                });
+
+                // After processing children, process the current node
+                if (node.id) {
+                    node.innerHTML = '';
+                    ctx.pantry.appendChild(node);
+                }
```

Comment:
if the current node has no id I think we have to remove it as I found otherwise non id elements remain when they shouldn't